src/components.js

```diff
@@ -501,45 +441,35 @@ export function parse(root) {
  */
 export function register(name, impl) {
 
-    var F, Surrogate;
+    var definition = {};
+    var htmlProto, proto;
 
-    if (isObject(name)) {
-        impl = name;
-        name = impl.name;
-    }
+    impl = impl || {};
 
-    if (!isString(name) || !name) {
-        throw Error('"' + name + '" is not a valid component name');
+    if ('extends' in impl) {
+        htmlProto = htmlPrototypes[impl.extends];
+        definition.extends = impl.extends;
     }
```

Comment:
Instead of extending using a string key, just provide a base object that it can acquire the prototype to extend from. This allows us to extend from other components or any other type of element. Like HTMLImageElement.